scripts/util/HelperNode.lua

```diff
@@ -0,0 +1,110 @@
+    if self.node and entityExists(self.node) then
+        unlink(self.node)
+        delete(self.node)
+    end
+end
+
+--- Place the node at the given position and rotation.
+---@param x number
+---@param y number if the node is linked to the terrain, this is relative to the terrain height at x, z
+---@param z number
+---@param yRotation number|nil Rotation set only if not nil
+function HelperNode:place(x, y, z, yRotation)
+    setTranslation(self.node, x, y, z)
+    if yRotation then
+        setRotation(self.node, 0, yRotation, 0)
+    end
+end
+
+---@param text string|nil
+function HelperNode:draw(text)
+    if entityExists(self.node) then
+        DebugUtil.drawDebugNode(self.node, text or getName(self.node), false, 0)
+    end
+end
+
+-- A helper node that is linked to the terrain root node, and the y coordinate is always
+-- the terrain height at the x, z position.
+---@class HelperTerrainNode : HelperNode
+HelperTerrainNode = CpObject(HelperNode)
+
+--- Create a new terrain helper node, linking it to the terrain root node.
+---@param name string
+function HelperTerrainNode:init(name)
+    HelperNode.init(self, name, g_currentMission.terrainRootNode)
+end
+
+--- Place the node at the given position and rotation.
+---@param x number
+---@param y number if the node is linked to the terrain, this is relative to the terrain height at x, z
+---@param z number
+---@param yRotation number|nil Rotation set only if not nil
+function HelperTerrainNode:place(x, y, z, yRotation)
+    setTranslation(self.node, x, getTerrainHeightAtWorldPos(g_currentMission.terrainRootNode, x, 0, z) + y, z)
+    if yRotation then
+        -- leave z/x rotation as is
+        local xRotation, _, zRotation = getWorldRotation(node)
+        setRotation(self.node, xRotation, yRotation, zRotation)
+    end
+end
+
+--- Place the node at the same world position and rotation as the given node.
+--- An optional local position can be given to offset the helper node from the given node.
+---@param node number
+---@param y number|nil relative height to terrain
+---@param lx number|nil x coordinate of the point relative to node
+---@param ly number|nil y coordinate of the point relative to node
+---@param lz number|nil z coordinate of the point relative to node
+function HelperTerrainNode:placeAtNode(node, y, lx, ly, lz)
+    local x, _, z = localToWorld(node, lx or 0, ly or 0, lz or 0)
+    local xRotation, yRotation, zRotation = getWorldRotation(node)
+    setRotation(self.node, xRotation, yRotation, zRotation)
+    self:place(x, y or 0, z)
+end
+
+--- Get the position of a point relative to node, in the helper node's coordinate system.
+---@param node number
+---@param lx number x coordinate of the point relative to the node
+---@param ly number y coordinate of the point relative to the node
+---@param lz number z coordinate of the point relative to the node
+---@return number, number, number
+function HelperTerrainNode:localToLocal(node, lx, ly, lz)
```

Comment:
This function is only working for HelperTerrainNode and not for the base HelperNode